### ISSUE TITLE
Multiple outputs in workload definition lead to "TraitDefinition AuxiliaryWorkload not found"

### ISSUE THREAD

**Issue by joscha-alisch** [oam-dev/kubevela#988]
**Describe the bug**
When creating a workload definition with multiple outputs as supported by #937 and uploading an app with that workload type, the kubevela controller fails creating that app with the error message 
```
kubevela-vela-core-6785bc9c79-dbnp9:2021-02-02T08:16:02.348Z    ERROR    Application    [Handle apply]    {"application": "
 default/reproduce-bug", "error": "admission webhook \"validating.core.oam.dev.v1alpha2.applicationconfigurations\" denied t
 he request: cannot get trait definition in component \"reproduce-bug\": TraitDefinition.core.oam.dev \"AuxiliaryWorkload\"
 not found"}
 ```

**To Reproduce**
1. Create a workload definition with multiple outputs, for example:
```yaml
apiVersion: core.oam.dev/v1alpha2
kind: WorkloadDefinition
metadata:
  name: multiple
  annotations:
    definition.oam.dev/description: A web service with multiple outputs
spec:
  definitionRef:
    name: deployments.apps
  extension:
    template: |
      output: {
          apiVersion: "apps/v1"
          kind:       "Deployment"
          spec: {
            selector: matchLabels: {
              "app.oam.dev/component": context.name
            }

            template: {
              metadata: labels: {
                "app.oam.dev/component": context.name
              }

              spec: {
                containers: [{
                  name:  context.name
                  image: "crccheck/hello-world"

                  ports: [{
                      containerPort: 8000
                      name: "web"
                  }]
                }]
            }
          }
        }
      }

      outputs: service: {
        apiVersion: "v1"
        kind:       "Service"
        metadata:
          name: context.name
        spec: {
          selector:
            "app.oam.dev/component": context.name
          ports: [
            {
              port:       80
              targetPort: "web"
            },
          ]
        }
      }

      parameter: {}
```
`kubectl apply -f multiple.yaml`

2. Create an app that references this workload:
```yaml
name: reproduce-bug
services:
  reproduce-bug:
    type: multiple
```
`vela up -f buggy-app.yaml` 

3. See that the resources don't get created
4. Look in the vela-core logs and see the following message:
```json
{"application": " default/reproduce-bug", "error": "admission webhook \"validating.core.oam.dev.v1alpha2.applicationconfigurations\" denied the request: cannot get trait definition in component \"reproduce-bug\": TraitDefinition.core.oam.dev \"AuxiliaryWorkload\" not found"}
```

**Expected behavior**
The app should be created with all outputs defined in the workload definition

**Cluster information**
Kubernetes Version: GKE v.1.19.6-gke.600
Kubevela Version: 0.3.2

**Comment by captainroy-hy** [comment]
Hi @joscha-alisch it's related to issue #919. I'm going to fix it.
Webhook should not fail if the traitdefinition is not found.

**Comment by captainroy-hy** [comment]
Currently, a possible workaround is disabling the webhook, you can follow [this doc ](https://github.com/oam-dev/kubevela/blob/master/docs/en/install.md#2-install-kubevela-controller)to setup vela without webhook enabled.

**Comment by joscha-alisch** [comment]
Thanks @captainroy-hy, we moved the service into a trait definition which is also fine as a workaround for now.

**Comment by captainroy-hy** [comment]
fixed in #991

**Comment by joscha-alisch** [comment]
Thanks @captainroy-hy , we can confirm that this works now :+1:
